admin-dev/themes/new-theme/js/product-page/combination.js

```diff
@@ -0,0 +1,123 @@
+import $ from 'jquery';
+
+export default function() {
+  $(document).ready(function() {
+    let $jsCombinationsList = $('.js-combinations-list');
+    let idsProductAttribute = $jsCombinationsList.data('ids-product-attribute').toString().split(',');
+    let idsCount = idsProductAttribute.length;
+    let currentCount = 0;
+    let step = 5;
+
+    $.get($jsCombinationsList.attr('data-action-refresh-images') + '/' + $jsCombinationsList.data('id-product'))
+      .then(function(response) {
+        getCombinations(response);
+        $('#create-combinations').click(function() {
+          generate();
+        });
+      });
+
+    let combinationUrl = $jsCombinationsList.data('combinations-url') + '/' + idsProductAttribute.slice(currentCount, currentCount+step).join('-');
+
+    let getCombinations = (combinationsImages) => {
+      $.get(combinationUrl).then(function (resp) {
+        $('#loading-attribute').before(resp);
+        refreshImagesCombination(combinationsImages, idsProductAttribute.slice(currentCount, currentCount+step));
+        currentCount += step;
+        combinationUrl = $jsCombinationsList.data('combinations-url') + '/' + idsProductAttribute.slice(currentCount, currentCount+step).join('-');
+        if (currentCount <= idsCount) {
+          getCombinations(combinationsImages);
+        } else {
+          $('#combinations-bulk-form').removeClass('inactive');
+          $('#loading-attribute').fadeOut(1000).remove();
+        }
+      });
+    };
+  });
+
+  let refreshImagesCombination = (combinationsImages, idsProductAttribute) => {
+    $.each(idsProductAttribute, function (index, value) {
+      var $combinationElem = $('.combination[data="' + value + '"]');
+      var $imagesElem = $combinationElem.find('.images');
+      var $index = $combinationElem.attr('data-index');
+
+      $imagesElem.html('');
+
+      $.each(combinationsImages[value], function(key, image) {
+        $imagesElem.append('<div class="product-combination-image ' + (image.id_image_attr ? 'img-highlight' : '') + '">\
```

Comment:
if it's `ES6` compliant you can use the [template literals](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals) here: what do you think ?